Dynamically size mod carousel

ShouvikGhosh2048 opened the pull request:
Closes #605.

otobot1 commented:
1. Image sizes can change even within the same mod and right now that leaves a gap below the images. I think the best option for handling this is to scale the carousel viewport so that it's always the same size as the active image, but I'm open to other suggestions. If we do it this way, we'll also want to ensure that the main carousel element's size stays constant for a particular mod so that things don't move around as the image sizes change.
![image](https://github.com/celestemods-com/CelesteMods/assets/27628434/ab1b8576-4983-4656-9d55-6939f151a6ce)
2. I'm seeing an error pop up in the browser console once I open an expanded mod. Please either fix the error, or determine that it's not caused by these changes.
![image](https://github.com/celestemods-com/CelesteMods/assets/27628434/3373f05f-5f14-480e-8c29-115714bad88f)
3. On larger mods, the carousel is not limited by vertical space yet isn't expanding horizontally as far as it can. For mods where there aren't vertical constraints, I would like the pictures to use all available horizontal space.
![image](https://github.com/celestemods-com/CelesteMods/assets/27628434/bc6cd5a7-6629-48b5-b665-26c48cfe1eff)
4. The mods table page is being reset when it shouldn't, but it's not obvious to me why that is happening. I doubt that this has to do with these changes, but please glance at this and see if you agree.

ShouvikGhosh2048 commented:
1) I've removed the border and aligned the images towards the center. Let me know how this feels.
2) This error was there before.
3) So images have a fixed width, independent of height. Do you want to change the width depending on the number of maps? If so by how much?
4) That was a bug in my earlier tabs PR - I was setting the selectedChildDifficulties as a new array even when it didn't change, which caused the page to reset. I've fixed that.

otobot1 commented:
1. 👍 
2. 👍 
3. I'll have a think about this and make a follow-up issue if needed.
4. 👍
